### ISSUE TITLE
Remove get_or_404 calls

### ISSUE THREAD

**Issue by glezos** [jtauber-archive/django-notification#29]
I suppose most of the users are treating django-notification as a library instead of a front-end, and its views/methods aren't really exposed to the user directly. There are calls of get_or_404 in the code which, if a notice type doesn't exist, triggers a 404 error. If the user (and developer) are showing a particular object (which does exist), and django-notification throws a 404 error for its own (valid) reasons, then this is confusing, and not easy to handle.

Bottom line: we should be throwing a NoticeTypeDoesNotExist exception or something.
